packages/keyring-controller/src/KeyringController.ts

```diff
@@ -697,7 +698,10 @@ export class KeyringController extends BaseController<
         },
       ),
     );
-    this.update({ keyrings: [...keyrings] });
+    this.update({
+      keyrings: [...keyrings],
+      vault: this.#keyring.store.getState().vault,
+    });
```

Comment:
Is there a moment where the inner vault changes but the method `fullUpdate` is not called?

Reply:
If all the interactions are done through `KeyringController` (which should be the case as it is inaccessible from the outside) then `fullUpdate` is called on each of those interactions, and both outer keyrings and vault should be updated along with the inner ones; all the interactions are covered as far as I know.

In [this refactor to BaseControllerV2](https://github.com/MetaMask/core/pull/1378) we start using event listeners instead, so we are sure that they are always updated synchronously; 
We can perhaps start doing it in this PR for what concerns the `vault`, putting it outside the `fullUpdate` method

Reply:
I moved the vault update to an event handler, it should now be updated everytime the inner persistent store is changed

Reply:
Looks better!